web-app/src/test/com.isa.aem/calculatorMethod/CurrencyExchangeRateRepositoryTest.java

```
@@ -0,0 +1,212 @@
+        // assert
+        assertThat(result)
+                .isBeforeOrEqualTo(LocalDate.now());
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    @DisplayName("Should return message \"No value present\" if currency is not on the list")
+
+    public void returnsThrowWhenCurrencyNotInTheFileForMethodGetMaxDateForSelectedCurrency() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "AA";
+        // act
+        currencyExchangeRateRepository.getMaxDateForSelectedCurrency(currency);
+
+    }
+
+    @Test
+    @DisplayName("Should return date after 1900-01-01 if currency is in File")
+
+    public void returnsMinDateForCorrectCurrencyName() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "PLN";
+        // act
+        LocalDate result = currencyExchangeRateRepository.getMinDateForSelectedCurrency(currency);
+        // assert
+        assertThat(result)
+                .isAfterOrEqualTo(LocalDate.of(1900, 01, 01));
+    }
+
+    @Test
+    @DisplayName("Should return date before 1900-01-01  when currency is in File even though it is write with a lowercase letter")
+
+    public void returnsMinDateForCorrectCurrencyWhenIsWriteInLowerCase() {
+        currencyExchangeRateRepository = new CurrencyExchangeRateRepository();
+        // arrange
+        String currency = "pln";
+        // act
+        LocalDate result = currencyExchangeRateRepository.getMinDateForSelectedCurrency(currency);
+        // assert
+        assertThat(result)
+                .isAfterOrEqualTo(LocalDate.of(1900, 01, 01));
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    @DisplayName("Should return \"No value present\" if currency is not in the File")
+
+    public void returnsThrowWhenCurrencyNotInTheFileForMethodGetMinDateForSelectedCurrency() {
```

nazwa tej metody jest niespojna z DisplayName i nieczytelna. to co jest w DisplayName jest bardziej czytelna ale to chyba nie jest to czego oczekujesz. z kodu wyglada na to ze oczekujesz exception a nie messsage "No value present"

---

ok. zmien tylko to co sie tyczy 'return throw'
napisalas, powinien "zwrocic rzucic kiedy cos tam"...sprobuj jedno z ponizszych: 

should throw an exception jakis tam kiedy cos tam
should return an exception jakis tam kiedy cos tam (czyli zwrocic wyjatek)